addons/mail/static/src/js/systray/systray_messaging_menu.js

```diff
@@ -324,6 +325,27 @@ var MessagingMenu = Widget.extend({
             }
         }
     },
+
+    /**
+     * When a expand is clicked on, we want to open the related object
+     *
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClickExpand: function (ev) {
+        ev.stopPropagation();
+        var $preview = $(ev.currentTarget).closest('.o_mail_preview');
+        var documentModel = $preview.data('documentModel');
+        var documentID = $preview.data('documentId');
+        if (documentID && documentModel) {
```

Comment:
I guess this guards prevents crash with mail failures. If you extend previews with `isLinkedToDocumentThread` property, I wouldn't have to do that, because the expand icon would be displayed only when necessary.